Factory Reset and RDM added

Clarence-F opened the pull request:
Trial addition of adding RDM and Factory Reset to the YAML

Commit:
Web API documentation and project configuration

Change-Id: Ie579d516d2eab0ed4e9b341c0da8487f2b5ab26a

Commit:
Add instructions for Windows to readme

Change-Id: I754295ef2a2c6f53b01748b85c7f3ab42aef67e5

Commit:
Fix path in readthedocs configuration

Change-Id: I52d6c8f38711cf3b1cd3d370e8b9ad35522b3cfd

Clarence-F committed:
Factory Reset and RDM added

Trial addition of adding RDM and Factory Reset to the YAML

sjmh3 commented on sphinx-src/web-api/factory-reset.rst:
```diff
@@ -0,0 +1,12 @@
+Factory Reset
+##############
+
+Reboots the controller and wipes all settings off the unit.
```
```suggestion
Deletes all controller settings and reboots the unit.
```

sjmh3 commented on sphinx-src/web-api/rdm.rst:
```diff
@@ -0,0 +1,104 @@
+RDM Discovery
+##############
```
```suggestion
#############
```

sjmh3 commented on sphinx-src/web-api/rdm.rst:
```diff
@@ -0,0 +1,104 @@
+.. _discover-http-post:
+
+POST
+===
```
```suggestion
====
```

sjmh3 commented on sphinx-src/web-api/rdm.rst:
```diff
@@ -0,0 +1,104 @@
+RDM Discovery
+##############
+
+Returns RDM data from fixtures, or triggers an RDM scan on the desired controller.
```
```suggestion
Discover RDM fixtures and get cached device info from discovered fixtures.
```

sjmh3 commented on sphinx-src/web-api/rdm.rst:
```diff
@@ -0,0 +1,104 @@
+HTTP
+****
+
+.. _discover-http-get:
```
Do you need this target? I generally only add these if they're needed.

sjmh3 commented on sphinx-src/web-api/rdm.rst:
```diff
@@ -0,0 +1,104 @@
+
+Returns an object with the same attributes as in the HTTP GET response.
+
+.. _discover-http-post:
```
Do you need this target?

sjmh3 commented on sphinx-src/web-api/rdm.rst:
```diff
@@ -0,0 +1,104 @@
+JavaScript
+**********
+
+??? - Not sure how to get this info...?
```
In the query.js file - I sent you v5 & v6 for you to compare.

sjmh3 commented on sphinx-src/web-api/rdm.rst:
```diff
@@ -0,0 +1,104 @@
+
+``GET /api/rdm/discovery``
+
+Returns a JSON object with the following attributes:
```
This is wrong, but it's wrong because the YAML is wrong. It returns an array of device info for a particular universe. YAML needs updating before you can progress.

sjmh3 commented on sphinx-src/web-api/rdm.rst:
```diff
@@ -0,0 +1,104 @@
+POST
+===
+
+``POST /api/rdm/discovery``
```
I show the query string params in this line, e.g. see the GET in web-api/output.rst

sjmh3 commented on sphinx-src/web-api/rdm.rst:
```diff
@@ -0,0 +1,104 @@
+===
+
+``POST /api/rdm/discovery``
+Triggers an RDM poll of the defined universe
```
See web-api/output.rst for the way to intro one of these GETs with a param in the query.

Clarence-F committed:
RDM Update

Update for the RDM Poll page

Clarence-F committed:
Update rdm.rst

Made some of Simon's changes

Clarence-F committed:
Project State, Space and XPC Mode

Added new pages for Space, Project State and the XPC Mode. All are Expert exclusive.

Clarence-F committed:
Update factory-reset.rst

Changed text as requested in review

Clarence-F committed:
??? to TODO

Changed all instances of ??? to TODO

Clarence-F committed:
Added setup page

Added setup page and updated index

Clarence-F commented:
I've changed the pull requests, so we can close this off now, I think...
